packages/legend-application-studio/src/stores/sidebar-state/BulkServiceRegistrationState.ts

```diff
@@ -0,0 +1,370 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { action, computed, makeObservable, observable, flow } from 'mobx';
+import type { EditorSDLCState } from '../EditorSDLCState.js';
+import type { EditorStore } from '../EditorStore.js';
+import {
+  type PureExecution,
+  areMultiplicitiesEqual,
+  buildLambdaVariableExpressions,
+  generateMultiplicityString,
+  Multiplicity,
+  VariableExpression,
+  ServiceExecutionMode,
+  type BulkRegistrationResultFail,
+  BulkRegistrationResultSuccess,
+  type BulkServiceRegistrationResult,
+} from '@finos/legend-graph';
+import {
+  type GeneratorFn,
+  ActionState,
+  assertNonEmptyString,
+  assertTrue,
+  filterByType,
+  getNullableFirstElement,
+  guaranteeNonNullable,
+  prettyCONSTName,
+  UnsupportedOperationError,
+  assertErrorThrown,
+  LogEvent,
+} from '@finos/legend-shared';
+import { Version } from '@finos/legend-server-sdlc';
+import { MINIMUM_SERVICE_OWNERS } from '../editor-state/element-editor-state/service/ServiceEditorState.js';
+import { ServiceRegistrationEnvironmentConfig } from '../../application/LegendStudioApplicationConfig.js';
+import { LEGEND_STUDIO_APP_EVENT } from '../LegendStudioAppEvent.js';
+import { generateServiceManagementUrl } from '../editor-state/element-editor-state/service/ServiceRegistrationState.js';
+
+export const LATEST_PROJECT_REVISION = 'Latest Project Revision';
+const getServiceExecutionMode = (mode: string): ServiceExecutionMode => {
+  switch (mode) {
+    case ServiceExecutionMode.FULL_INTERACTIVE:
+      return ServiceExecutionMode.FULL_INTERACTIVE;
+    case ServiceExecutionMode.SEMI_INTERACTIVE:
+      return ServiceExecutionMode.SEMI_INTERACTIVE;
+    case ServiceExecutionMode.PROD:
+      return ServiceExecutionMode.PROD;
+    default:
+      throw new UnsupportedOperationError(
+        `Encountered unsupported service execution mode '${mode}'`,
+      );
+  }
+};
+interface ServiceVersionOption {
+  label: string;
+  value: Version | string;
+}
+interface ServiceRegistrationResult {
+  successfulServices: string[];
+  failedServices: string[];
+  serviceLinks: string[];
+}
+export class ServiceConfigState {
```

Comment:
Can we think of a way to organize this model in conjuntion with `ServiceRegistrationState`. 
we may need one state to hold the config and both the serice element editor use it and the build registration one use it as well. 
something like maybe 
`ServiceRegistrationConfigState`
`BuildServiceRegistrationConfigState extends ServiceRegistrationConfigState`
`ServiceEditorServiceRegistrationConfigState extends ServiceRegistrationConfigState`
We want to minimize the repeated code in both states and think about what we share that is only relevant to the registration config